consensus/common/src/main/java/tech/pegasys/pantheon/consensus/common/jsonrpc/AbstractGetSignerMetricsMethod.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.consensus.common.jsonrpc;
+
+import tech.pegasys.pantheon.consensus.common.VoteTallyCache;
+import tech.pegasys.pantheon.ethereum.core.Address;
+import tech.pegasys.pantheon.ethereum.core.BlockHeader;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.JsonRpcRequest;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.parameters.JsonRpcParameter;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.queries.BlockchainQueries;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcError;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcErrorResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.results.report.block.production.ProposerReportBlockProductionResult;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.results.report.block.production.SignerMetricsResult;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.results.report.block.production.ValidatorReportBlockProductionResult;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.LongStream;
+
+public abstract class AbstractGetSignerMetricsMethod {
+
+  private final BlockchainQueries blockchainQueries;
+  private final VoteTallyCache voteTallyCache;
+  private final JsonRpcParameter parameters;
+
+  public AbstractGetSignerMetricsMethod(
+      final BlockchainQueries blockchainQueries,
+      final VoteTallyCache voteTallyCache,
+      final JsonRpcParameter parameter) {
+    this.blockchainQueries = blockchainQueries;
+    this.voteTallyCache = voteTallyCache;
+    this.parameters = parameter;
+  }
+
+  public JsonRpcResponse response(final JsonRpcRequest request) {
+
+    final long startBlock = parameters.required(request.getParams(), 0, Long.class);
+    final long endBlock = parameters.required(request.getParams(), 1, Long.class);
```

Comment:
We should support the usual values for block parameters like `latest`.  `parameters().required(request.getParams(), 0, BlockParameter.class);` will handle the parsing of that.  `pending` and `latest` would mean the same thing (current chain head) in this context.

I'd probably make both arguments optional as well.  Start would default to 100 blocks before current chain head and end defaults to chain head.  That way you can call with no arguments and get a useful result (most of the time you want to know which of the current validators hasn't proposed recently), or specify a starting block to get from their to chain head easily, or both start and end for a custom range.